lib/rules/no-implied-eval.js

```diff
@@ -28,141 +34,131 @@ module.exports = {
     },
 
     create(context) {
-        const CALLEE_RE = /^(setTimeout|setInterval|execScript)$/u;
-
-        /*
-         * Figures out if we should inspect a given binary expression. Is a stack
-         * of stacks, where the first element in each substack is a CallExpression.
-         */
-        const impliedEvalAncestorsStack = [];
-
-        //--------------------------------------------------------------------------
-        // Helpers
-        //--------------------------------------------------------------------------
+        const EVAL_LIKE_FUNCS = Object.freeze(["setTimeout", "execScript", "setInterval"]);
+        const GLOBAL_CANDIDATES = Object.freeze(["global", "window"]);
 
         /**
-         * Get the last element of an array, without modifying arr, like pop(), but non-destructive.
-         * @param {Array} arr What to inspect
-         * @returns {*} The last element of arr
-         * @private
+         * Checks whether a node is evaluated as a string or not.
+         * @param {ASTNode} node A node to check.
+         * @returns {boolean} True if the node is evaluated as a string.
          */
-        function last(arr) {
-            return arr ? arr[arr.length - 1] : null;
+        function isEvaluatedSting(node) {
+            if (
+                (node.type === "Literal" && typeof node.value === "string") ||
+                node.type === "TemplateLiteral"
+            ) {
+                return true;
+            }
+            if (node.type === "BinaryExpression" && node.operator === "+") {
```

Comment:
I wonder about some other cases that can result in string coercion like:

```js
String(value);
value.toString();
JSON.stringify(val);
```

Is there a way to reliably enforce these cases? We should be able to check if they're the globals for `String` and `JSON`, but I don't know if `X.prototype.toString()` can be reliably checked (though it might be a fair assumption to assume it returns a string, given the name).

Reply:
@kaicataldo 
 I'll try but I'm not sure to do that.  Can I work on that in another PR?

Reply:
I think it would be fine to do that, since this does already improve the rule. Let's see what others have to say.

Reply:
I'd also agree that this could be a nice future improvement.

Maybe we could also use [getstaticvalue](https://eslint-utils.mysticatea.dev/api/ast-utils.html#getstaticvalue) to catch code like:

```js
var s = "alert('Hi!');";
setTimeout(s, 100);
```

And perhaps add `self` to the list of global objects.